lib/models/watch-detector.js

```diff
@@ -0,0 +1,238 @@
+'use strict';
+
+var quickTemp = require('quick-temp');
+var Promise = require('../ext/promise');
+var WATCHMAN_INFO = 'Visit http://ember-cli.com/user-guide/#watchman for more info.';
+var semver = require('semver');
+var path = require('path');
+
+var POLLING = 'polling';
+var WATCHMAN = 'watchman';
+var NODE = 'node';
+
+var debug   = require('debug')('ember-cli:watcher');
+
+function WatchPreference(watcher) {
+  this.watcher = watcher || null;
+  this._watchmanInfo = {
+    enabled: false,
+    version: null,
+    canNestRoots: false
+  };
+}
+
+WatchPreference.prototype.watchmanWorks = function(details) {
+  this._watchmanInfo.enabled = true;
+  this._watchmanInfo.version = details.version;
+  this._watchmanInfo.canNestRoots = semver.satisfies(details.version, '>= 3.7.0');
+};
+
+Object.defineProperty(WatchPreference.prototype, 'watchmanInfo', {
```

Comment:
What do we gain by adding this as a getter?  Just the ability to change in the future?